sdk/go/pulumi/rpc.go

```diff
@@ -63,14 +63,68 @@ func mapStructTypes(from, to reflect.Type) func(reflect.Value, int) (reflect.Str
 	}
 }
 
+type urnSet map[URN]struct{}
+
+func (s urnSet) add(v URN) {
+	s[v] = struct{}{}
+}
+
+func (s urnSet) has(v URN) bool {
+	_, ok := s[v]
+	return ok
+}
+
+func (s urnSet) union(other urnSet) {
+	for v := range other {
+		s.add(v)
+	}
+}
+
+func (s urnSet) values() []URN {
```

Comment:
Might sort it to make tests deterministic/stable.

Reply:
I went back and forth on that. All consumers of this will sort the list, so I didn't sort it here, but there's not really any harm in doing so. WDYT?